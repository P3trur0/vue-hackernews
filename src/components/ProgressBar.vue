<template>
  <div class="progress" :style="{
    'width': `${percent}%`
  }"></div>
</template>

<script>
export default {
  name: 'progress-bar',
  data: () => ({
    percent: 0
  })
}
</script>

<style scoped>
.progress {
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  height: 2px;
  width: 0%;
  transition: width 0.2s, opacity 0.4s;
  opacity: 0;
  background-color: #ffca2b;
  z-index: 999999;
}
.show {
  opacity: 1;
}
.error {
  background-color: #ff0000;
}
</style>
