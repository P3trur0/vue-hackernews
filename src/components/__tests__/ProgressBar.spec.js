import { mount } from 'vue-test-utils'
import ProgressBar from '../ProgressBar.vue'

describe('ProgressBar.vue', () => {
  test('has no show class onn initial render', () => {
    const wrapper = mount(ProgressBar)
    expect(wrapper.classes()).not.toContain('show') // #A
  })

  test('initializes with 0% width', () => {
    const wrapper = mount(ProgressBar)
    expect(wrapper.element.style.width).toBe('0%') // #A
  })

  test('sets .progress width as same value of vm.percentage', () => {
    const wrapper = mount(ProgressBar)
    const percent = 70
    wrapper.setData({ percent }) // #A
    expect(wrapper.element.style.width).toBe(`${percent}%`)// #B
  })
})
