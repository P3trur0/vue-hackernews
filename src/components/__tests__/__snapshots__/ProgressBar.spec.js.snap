// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar.vue renders in a success state when finish is called 1`] = `<div class="progress" style="width: 100%;"></div>`;

exports[`ProgressBar.vue renders in an error state when fail  is called 1`] = `<div class="progress error show" style="width: 10%;"></div>`;

exports[`ProgressBar.vue renders in an in progress state when start is called 1`] = `<div class="progress show" style="width: 10%;"></div>`;
